platform-metrics/build.gradle.kts

```diff
@@ -26,6 +26,7 @@ dependencies {
   implementation("io.prometheus:simpleclient_pushgateway:0.12.0")
   implementation("org.eclipse.jetty:jetty-servlet:9.4.53.v20231009")
   implementation("com.google.guava:guava:32.0.1-jre")
+  implementation("com.github.ben-manes.caffeine:caffeine:3.1.8")
```

Comment:
Have we started introducing caffeine in the platform?

Assuming yes, do we need the dependency here? This will add caffeine to the classpath of all projects regardless of whether they use it or not. Suggest leaving this as compileOnly. If anyone is calling it, they have a caffeine cache and should be providing the dependency at runtime.

Reply:
We want to use in detector for the first time to try out an approach for optimisation (behind a config).
Changed to compileOnly.